File: example/src/main/java/eu/darken/mvpbakery/example/ActivityBinderModule.java
package eu.darken.mvpbakery.example;

import dagger.Binds;
import dagger.Module;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;
import eu.darken.mvpbakery.example.screens.MainActivity;
import eu.darken.mvpbakery.example.screens.MainComponent;
import eu.darken.mvpbakery.injection.activity.ActivityKey;

@Module(subcomponents = {MainComponent.class})
abstract class ActivityBinderModule {

    @Binds
    @IntoMap
    @ActivityKey(MainActivity.class)
    abstract AndroidInjector.Factory<? extends android.app.Activity> mainActivity(MainComponent.Builder impl);
}

File: example/src/main/java/eu/darken/mvpbakery/example/ActivityBinderModule.kt
package eu.darken.mvpbakery.example

import android.app.Activity
import dagger.Binds
import dagger.Module
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap
import eu.darken.mvpbakery.example.screens.MainActivity
import eu.darken.mvpbakery.example.screens.MainComponent
import eu.darken.mvpbakery.injection.activity.ActivityKey

@Module(subcomponents = [MainComponent::class])
internal abstract class ActivityBinderModule {

    @Binds
    @IntoMap
    @ActivityKey(MainActivity::class)
    internal abstract fun mainActivity(impl: MainComponent.Builder): AndroidInjector.Factory<out Activity>
}